pkg/database/openshiftclusters.go

```diff
@@ -284,8 +284,11 @@ func (c *openShiftClusters) EndLease(ctx context.Context, key string, provisioni
 		if provisioningState == api.ProvisioningStateSucceeded {
 			doc.Dequeues = 0
 		}
-
-		doc.AsyncOperationID = ""
+		// If EndLease is called while cluster is still in non-terminal phase,
+		// we clean AsyncOperationID as this is just handover between phases
+		if provisioningState.IsTerminal() {
+			doc.AsyncOperationID = ""
+		}
```

Comment:
still need to reset dequeues on !failed

Reply:
And at this point dequeue is lost its purpose in my head. I might not get it now.  
We reset it at Success and anything else what is not Failed. And we not dequeue if its Failed. So we basically dequeue forever if there is logical error?

Reply:
if the backend crashes, it is not reset.

Reply:
So is dequeue only purpose is backend failures? and it is not indended to catch any logical errors?